### PR TITLE
refactor: move stackset logic out of deploy into its own pkg

### PR DESCRIPTION
**Pull request by efekarakus**
With this change, we finally got to extract all the cfn logic into its own package.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by efekarakus** [comment]
The e2e tests pass:
```
Ginkgo ran 5 suites in 1h49m45.5653426s
Test Suite Passed
```

File: internal/pkg/aws/cloudformation/stackset/stackset.go [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+func (ss *StackSet) CreateInstancesAndWait(name string, accounts, regions []string) error {
+	id, err := ss.createInstances(name, accounts, regions)
+	if err != nil {
+		return err
+	}
+	return ss.waitForOperation(name, id)
+}
+
+// InstanceSummariesOption allows to filter instance summaries to retrieve for the stack set.
+type InstanceSummariesOption func(input *cloudformation.ListStackInstancesInput)
```
Comment (kohidave) [inline internal/pkg/aws/cloudformation/stackset/stackset.go]: Neat!

File: internal/pkg/aws/cloudformation/stackset/stackset.go [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+		case *cloudformation.UpdateStackSetInput:
+			{
+				v.ExecutionRoleName = aws.String(roleName)
+			}
+		}
+	}
+}
+
+// WithAdministrationRoleARN sets an administration role arn for a stack set.
+func WithAdministrationRoleARN(roleARN string) CreateOrUpdateOption {
```
Comment (kohidave) [inline internal/pkg/aws/cloudformation/stackset/stackset.go]: You know what - I love this pattern now.  I am reformed.

File: internal/pkg/deploy/cloudformation/cloudformation_test.go [hunk 3]
```diff
@@ -4,64 +4,14 @@
...
-	"github.com/aws/aws-sdk-go/service/cloudformation"
-	"github.com/aws/aws-sdk-go/service/cloudformation/cloudformationiface"
 	"github.com/gobuffalo/packd"
 )
 
 const (
 	mockTemplate = "mockTemplate"
 )
 
-type mockCloudFormation struct {
```
Comment (kohidave) [inline internal/pkg/deploy/cloudformation/cloudformation_test.go]: 😍

File: internal/pkg/deploy/cloudformation/project.go [hunk 4]
```diff
@@ -120,36 +108,32 @@ func (cf CloudFormation) getResourcesForStackInstances(project *archer.Project,
...
 
-	stackInstances, err := cf.sdkClient.ListStackInstances(listStackInstancesInput)
-
+	summaries, err := cf.projectStackSet.InstanceSummaries(projectConfig.StackSetName(), opts...)
 	if err != nil {
-		return nil, fmt.Errorf("listing stack instances: %w", err)
+		return nil, err
 	}
-
-	regionalResources := []*archer.ProjectRegionalResources{}
```
Comment (kohidave) [inline internal/pkg/deploy/cloudformation/project.go]: Tyrant.

File: internal/pkg/aws/cloudformation/stackset/description.go [hunk 5]
```diff
@@ -0,0 +1,11 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package stackset
+
+// Description represents a created stack set resource.
```
Comment (iamhopaul123) [inline internal/pkg/aws/cloudformation/stackset/description.go]: nit: StackSet

File: internal/pkg/aws/cloudformation/stackset/stackset.go [hunk 6]
```diff
@@ -0,0 +1,353 @@
...
+		if err != nil {
+			return fmt.Errorf("delete stack instances in regions %v for accounts %v for stackset %s: %w",
+				regions, accounts, name, err)
+		}
+		if err := ss.waitForOperation(name, aws.StringValue(operation.OperationId)); err != nil {
+			return err
+		}
+	}
+
+	// Delete the stack set now that the stack instances are gone.
```
Comment (iamhopaul123) [inline internal/pkg/aws/cloudformation/stackset/stackset.go]: same here

**Pull request merged**